src/hidra/receiver/plugins/asapo_producer.py

```diff
@@ -264,29 +278,14 @@ def _callback(self, header, err):
         self.lock.acquire()
         header = {key: val for key, val in header.items() if key != 'data'}
         if err is None:
-            self.log.debug("Successfully sent: %s", header)
+            logger.debug("Successfully sent: %s", header)
         else:
-            self.log.error("Could not sent: %s, %s", header, err)
+            logger.error("Could not sent: %s, %s", header, err)
         self.lock.release()
 
     def _parse_file_name(self, path):
-        # check for ignored files
-        if re.search(self.ignore_regex, path):
-            raise Ignored("Ignoring file {}".format(path))
-
-        # parse file name
-        search = re.search(self.file_regex, path)
-        if search:
-            matched = search.groupdict()
-        else:
-            self.log.debug("file name: %s", path)
-            self.log.debug("file_regex: %s", self.file_regex)
-            raise utils.UsageError("Does not match file pattern")
-
-        data_source = self.data_source
-        if self.data_source is None:
-            data_source = get_entry(matched, "data_source")
-
+        matched = parse_file_path(self.file_regex, path)
+        data_source = get_entry(matched, "data_source", self.data_source)
```

Comment:
To avoid the tricky behavior of the default value in `get_entry`, I would write this explicitly as
```suggestion
        data_source = matched.get("data_source", self.data_source)
        if data_source is None:
            raise utils.UsageError(...)
```

Reply:
Changed.